src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Validation/ValidationVisitor.cs

```diff
@@ -192,6 +223,29 @@ protected virtual bool Visit(ModelMetadata metadata, string key, object model)
                 return true;
             }
 
+            if (MaxValidationDepth != null && CurrentPath.Count > MaxValidationDepth)
+            {
+                // Non cyclic but too deep an object graph.
+                
+                // Pop the current model to make ValidationStack.Dispose happy
+                CurrentPath.Pop(model);
+
+                string message;
```

Comment:
Add a comment that `ModelMetadataKind.Parameter` isn't possible because `MaxValidationDepth == 0` is disallowed.

Reply:
Ehh, doesn't really add value to the code.

Reply:
Every other `switch` on `MetadataKind` covers the three `ModelMetadataKind`. Want future developers to understand why this one doesn't without lots of head-scratching.

Reply:
Haven't addressed this and new comment doesn't say anything.